task7/src/main/scala/com/bootcamp/task/Homework.scala

```diff
@@ -0,0 +1,109 @@
+package com.bootcamp.task
+
+object Homework {
+	object Task1 {
+		final case class Money(amount: BigDecimal)
+
+		// TODO: create Ordering instance for Money
+		implicit val moneyOrdering: Ordering[Money] = (a,b) => a.amount compareTo b.amount
+	}
+
+	object Task2 {
+		trait Show[T] { // fancy toString
+			def show(entity: T): String
+		}
+
+		final case class User(id: String, name: String)
+
+		object ShowSyntax {
+			implicit class ShowOps[A](value: A) {
+				def show(implicit s: Show[A]): String = s.show(value)
+			}
+		}
+
+		import ShowSyntax._
+		implicit val showUser: Show[User] = user => s"My name is ${user.name} with ID ${user.id}"
+
+		User("1", "Oleg").show
+	}
+
+	object Task3 {
+		type Error = String
+		trait Parse[T] { // invent any format you want or it can be csv string
+			def parse(entity: String): Either[Error, T]
+		}
+
+		object CsvParseSpec {
```

Comment:
Why spec?

Reply:
I was thinking of something else, ment Syntax :)